src/tools/clippy/clippy_utils/src/lib.rs

```diff
@@ -172,10 +172,10 @@ pub fn expr_or_init<'a, 'b, 'tcx: 'b>(cx: &LateContext<'tcx>, mut expr: &'a Expr
 pub fn find_binding_init<'tcx>(cx: &LateContext<'tcx>, hir_id: HirId) -> Option<&'tcx Expr<'tcx>> {
     let hir = cx.tcx.hir();
     if_chain! {
-        if let Some(Node::Pat(pat)) = hir.find(hir_id);
+        if let Node::Pat(pat) = hir.get(hir_id);
```

Comment:
Looking through the Clippy changes, it seems that `get` asserts that this node exists. I'm not 100% sure that this will just work in all cases in Clippy.

So if your assertion that all nodes exist by construction is true this should be fine. Otherwise I can see many new ICE issues in the future. I'm not informed enough to help with a decision regarding this, though.